packages/override-tools/src/ActionQueue.ts

```diff
@@ -0,0 +1,36 @@
+/**
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT License.
+ *
+ * @format
+ */
+
+/**
+ * Executes actions sequentially, fullfilling the returned promise once complete.
+ */
+export default class ActionQueue<T> {
+  private actions: Array<() => Promise<void>> = [];
+
+  enqueue(action: () => Promise<T>): Promise<T> {
+    return new Promise((resolve, reject) => {
+      this.actions.push(async () => {
```

Comment:
>async [](start = 24, length = 5)

Is this eneded here? actions are defined without "async" on line 12

Reply:
Yep, all we do here is queue the item, kick off a separate async function to pump queue items, then return immediately. This is because we don't want to execute anything concurrently.